### PR TITLE
Fix/search bug

### PR DESCRIPTION
**Pull request by Matt-Roden**
# Description drawing

- Added foundPokemon into the second argument of useEffect on the PokemonDetails page. 
- Because of how useEffect works, with an empty array as the second argument, we were telling it to `do the thing` only once, which would be on page load. Based on how we currently have our app set up, when a user searches for a new pokemon, the details component does not bring the user to a new page, which would cause useEffect to fire again. Instead it just renders the details component onto the home screen. The only thing that updates in the details component is the foundPokemon prop being passed down, so by plugging it into the second argument of useEffect, we are now triggering useEffect on every update of the prop being passed down to it. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] local host 3000
- [x] console.log()
- [x] dev tools
- [ ] cypress testing

# Checklist:

- [x] My code follows the style guidelines of this project/linted
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/PokemonDetails/PokemonDetails.js [hunk 1]
```diff
@@ -23,15 +21,10 @@ const PokemonDetails = ({ foundPokemon, getPokemonImage }) => {
       setError(error.message);
     }
   };
-// for some reason I don't know yet --it is not reseting the state when we type a diferent name or id 
-// i'm not sure if only clearing the search it is what we need, probably we need a way to look into the details again..
   useEffect(() => {
     getPokemonDetails();
-  }, []);
+  }, [foundPokemon]);
```
Comment (hayleyw7) [inline src/components/PokemonDetails/PokemonDetails.js:26]: THIS DID IT!!!!!!!!

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -2,48 +2,48 @@ import React, { Component } from 'react';
...
-        // console.log(queriedPokemon)
-        this.props.addPokemon(queriedPokemon)
-        // if(props.addPokemon(queriedPokemon)
-        // then addPokemon(queriedPokemon)
-        // else show error
+  handleClick = (e) => {
+    e.preventDefault();
+    const queriedPokemon = {
+        key: Date.now(),
+        ...this.state
```
Comment (hayleyw7) [inline src/components/Search/Search.js:21]: Thanks for cleaning up the code.

**Review by hayleyw7 — APPROVED** [review]
You did it! Great job!

File: src/components/App/App.js [hunk 3]
```diff
@@ -24,13 +23,12 @@ class App extends Component {
       });
     });
   }
-  // i think we will modify this function ? or maybe not just add a function inside that calls the rigth api
+
```
Comment (joanafbrito) [inline src/components/App/App.js:28]: I like that you are cleaning the comments after we were able to figure this out!

File: src/components/PokemonDetails/PokemonDetails.js (hunk 1 above)
Comment (joanafbrito) [reply src/components/PokemonDetails/PokemonDetails.js:26]: Awesome!!

**Review by joanafbrito — APPROVED** [review]
Thank you for cleaning up and for fix that bug

**Pull request merged**